### PR TITLE
skupper configures closeConnectionsOnDelete flag

### PR DESCRIPTION
**Pull request by Karen-Schoener**

### CODE REVIEW

File: pkg/qdr/qdr.go [hunk 1]
```diff
@@ -501,13 +505,14 @@ type Address struct {
...
-	SslProfile     string `json:"sslProfile,omitempty"`
-	VerifyHostname *bool  `json:"verifyHostname,omitempty"`
+	Name                     string `json:"name,omitempty"`
+	Host                     string `json:"host,omitempty"`
+	Port                     string `json:"port,omitempty"`
+	Address                  string `json:"address,omitempty"`
+	SiteId                   string `json:"siteId,omitempty"`
+	SslProfile               string `json:"sslProfile,omitempty"`
+	VerifyHostname           *bool  `json:"verifyHostname,omitempty"`
+	CloseConnectionsOnDelete *bool  `json:"closeConnectionsOnDelete,omitempty"`
```
Comment (gabordozsa) [inline pkg/qdr/qdr.go:515]: Could the type be `bool` instead of `*bool` ?  That would simplify the rest of the code.

**Comment by grs** [comment]
Is there a use-case for configuring this per tcpConnector? Why would you want one behaviour for one connector and the other behaviour for another?

**Comment by gabordozsa** [comment]
> Is there a use-case for configuring this per tcpConnector? Why would you want one behaviour for one connector and the other behaviour for another?

The new flag is defined per tcpConnector and tcpListener in the router. This was suggested by the router team as opposed to have a global router config flag.  The justification was that router code changes would be more or less the same.    I'm not aware of any use case though for turning the flag on for some tcpListeners and tcpConnectors but not for some others. In fact, from the MESH project perspective, we need a way to set this flag globally for skupper (e.g. via a new` skupper init ` flag as @Karen-Schoener suggested before).

File: pkg/qdr/qdr.go (hunk 1 above)
Comment (Karen-Schoener) [reply pkg/qdr/qdr.go:515]: Ah, I see that I misread the corresponding skupper-router PR.  

When I saw this line in the skupper-router PR, I (incorrectly) assumed that the new optional flag was a bool pointer.
My mistake.
` config->terminate_conns   = qd_entity_opt_bool(entity, "closeConnectionsOnDelete", false);`
 
I will update to change CloseConnectionsOnDelete to be a bool.

**Comment by Karen-Schoener** [comment]
Closing in favor of https://github.com/skupperproject/skupper/pull/1564

**Pull request closed without merging**